### PR TITLE
[doc][client feature matrix] improve experience for .NET users looking for information on .NET/C# clients

### PR DESCRIPTION
**Pull request by jak78**
This PR aims to improve the experience for .NET/C# users looking for information on .NET/C# clients. The goal is to help the users choose which client best suits their needs.

Two .NET clients exist: C#/DotPulsar and .NET client. They don't provide the same feature coverage. The PR fixes the client features matrix to display the two clients side by side to highlight this coverage difference.

Additionally, the PR adds a link to the client docs to facilitate the navigation and help the user compare the two clients in details.

Finally, the PR makes the doc use the same names for the clients in both the feature matrix and the client libraries page to improve clarity.

Preview:

Screenshot 2023-12-18 at 3 33 49 PM

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: client-feature-matrix/index.mdx [hunk 1]
```diff
@@ -14,6 +14,8 @@ import {
   tableview,
 } from "../data/matrix.js";
 
+To get more information on clients, see [Pulsar Client Libraries](/docs/next/client-libraries).
```
Comment (tisonkun) [inline client-feature-matrix/index.mdx:17]:
```suggestion
To get more information on clients, see [Pulsar Client Libraries](/docs/client-libraries).
```

Let's use latest stable version.

Comment (jak78) [reply client-feature-matrix/index.mdx:17]: I tried that, but it results in a 404 error

Comment (jak78) [reply client-feature-matrix/index.mdx:17]: Looks like you can remove the `next` part from the links on the `md` pages but not on the `mdx` pages

Comment (tisonkun) [reply client-feature-matrix/index.mdx:17]: > it results in a 404 error

The redirect happens in Apache Web Server. So perhaps you need run docker-compose for previewing this manner ..

https://github.com/apache/pulsar-site/blob/80e4090e04dbbbf22652be38d542e4989d8dc3bd/static/.htaccess#L19

https://github.com/apache/pulsar-site/blob/main/docker-compose.yaml

Comment (jak78) [reply client-feature-matrix/index.mdx:17]: Same result using docker compose

Comment (tisonkun) [reply client-feature-matrix/index.mdx:17]: Well. I'll merge as is now.

**Comment by jak78** [comment]
Thank you @tisonkun

**Pull request merged**